tests/mock_unit/test_oob_telemetry.py

```diff
@@ -139,28 +146,38 @@ def test_unit_oob_log_not_implemented_error(caplog, local_testing_telemetry_setu
 
     # test sending successfully
     caplog.clear()
-    with caplog.at_level(logging.DEBUG, logger="snowflake.snowpark.mock._telemetry"):
-        oob_service.log_not_supported_error(
-            external_feature_name=unraise_feature_name,
-        )
+
+    max_retry = 10
```

Comment:
What about replacing this with a pytest plugin: https://github.com/box/flaky?

Reply:
this is very cool ^

Reply:
this is good idea, but I did this retry manually for 2 reasons here:

1. I'm not sure introducing a new dependency (though just for test) would require any stored proc test change or other configure change. luckily nothing breaks and works, but we need extra time to test.
2. though it's test dependency, does it still need 3rd party library review process?

so I think we can log a JIRA for improvement at a later time.